crherman7/rechunk
# feat: rollup preset

## Scope



[closes TICKET-22](https://github.com/crherman7/rechunk/issues/22)

## Implementation

Creates a Rollup preset. This includes all the required rollup plugins to generate a clean bundle that can be used by ReChunk to render a string to a component. It additionally modifies the babel configuration to exclude module resolver plugin and set enableBabelRuntime to false for the react native babel preset.

## Screenshots

<details>
  <summary>ios</summary>

![Simulator Screenshot - iPhone 15 - 2024-08-29 at 18 33 51](https://github.com/user-attachments/assets/278508ae-a056-49b6-97d6-837279aa463b)
</details>

<details>
  <summary>android</summary>
  
![Screenshot_1724971045](https://github.com/user-attachments/assets/cc7cc6cd-be0e-40ed-ab4b-00cbce109884)
</details>

## How to Test

Run the example app for both platforms and verify the ReChunk components render.

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |
